Src/Support/Google.Apis.Auth/OAuth2/AwsExternalAccountCredential.AwsSecurityCredential.cs

```diff
@@ -0,0 +1,144 @@
+                    $"At least one of the following environment variables was not set: {AccessKeyIdEnvVar}, {SecretAccessKeyEnvVar}." +
+                    $"And the {nameof(SecurityCredentialsUrl)} was not specified in the credential configuration" +
+                    "or no security credentials couldn't be obtained from it.");
+
+            /// <summary>
+            /// Attempts to fetch the security credentials from environment variables.
+            /// Returns null if the environment variables are not set.
+            /// </summary>
+            internal static AwsSecurityCredentials MaybeFromEnvVars()
+            {
+                if (AccessKeyIdEnvVar.GetNonEmptyEnvVarValue() is string accessKeyId
+                    && SecretAccessKeyEnvVar.GetNonEmptyEnvVarValue() is string secretAccessKey)
+                {
+                    return new AwsSecurityCredentials(accessKeyId, secretAccessKey, TokenEnvVar.GetNonEmptyEnvVarValue());
+                }
+
+                return null;
+            }
+
+            /// <summary>
+            /// Attempts to fetch the security credentials from the metadata server.
+            /// Returns null if the credential URL is null or empty.
+            /// </summary>
+            internal static async Task<AwsSecurityCredentials> MaybeFromMetadataAsync(AwsMetadataServerClient metadataClient, string credentialUrl)
+            {
+                // https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#instance-metadata-security-credentials
+                if (!string.IsNullOrEmpty(credentialUrl)
+                    // Build the actual credential endpoint.
+                    && await BuildSecurityCredentialsEndpointAsync().ConfigureAwait(false) is string credentialEndpoint
+                    // Fetch the credential JSON
+                    && await metadataClient.FetchMetadataAsync(credentialEndpoint).ConfigureAwait(false) is string credentialJson
+                    && credentialJson != "")
+                {
+                    // Deserialize the credentials
+                    var deserializedCredentials = NewtonsoftJsonSerializer.Instance.Deserialize<AwsSecurityCredentialsResponse>(credentialJson);
+
+                    if (deserializedCredentials.IsSuccess
+                        && !string.IsNullOrEmpty(deserializedCredentials.AccessKeyId)
+                        && !string.IsNullOrEmpty(deserializedCredentials.SecretAccessKey))
+                    {
+                        return new AwsSecurityCredentials(
+                            deserializedCredentials.AccessKeyId, deserializedCredentials.SecretAccessKey,
+                            deserializedCredentials.Token == "" ? null : deserializedCredentials.Token);
+                    }
+
+                    throw new InvalidOperationException(
+                        $"The request to the metadata server security credentials endpoint {credentialEndpoint} was not succesful." +
```

Comment:
succesful => successful, space at end of line, and either change the period to a comma, or capitalize T on the next line.